src/python/ASO/Rucio/Actions/RegisterReplicas.py

```diff
@@ -0,0 +1,278 @@
+        new dataset when the current dataset exceeds
+        `config.arg.max_file_per_datset`.
+
+        :param replicas: a list contain dict with REST xfer id and replicas LFN.
+        :type replicas: list
+        """
+
+        containerReplicas = []
+        newReplicas = []
+        # filter out duplicated replicas
+        replicasInContainer = self.transfer.replicasInContainer[container]
+        for r in replicas:
+            if r['name'] in replicasInContainer:
+                c = r.copy()
+                c['dataset'] = replicasInContainer[r['name']]
+                containerReplicas.append(c)
+            else:
+                newReplicas.append(r)
+
+        b = BuildDBSDataset(self.transfer, self.rucioClient)
+        currentDataset = b.getOrCreateDataset(container)
+        self.logger.debug(f'currentDataset: {currentDataset}')
+        for chunk in chunks(newReplicas, config.args.replicas_chunk_size):
+            dids = [{
+                'scope': self.transfer.rucioScope,
+                'type': "FILE",
+                'name': x["name"]
+            } for x in chunk]
+            # no need to try catch for duplicate content. Not sure if
+            # restart process is enough for the case of connection error
+            attachments = [{
+                'scope': self.transfer.rucioScope,
+                'name': currentDataset,
+                'dids': dids
+            }]
+            self.rucioClient.add_files_to_datasets(attachments, ignore_duplicate=True)
+            successItems = [{
+                'id': x['id'],
+                'name': x['name'],
+                'dataset': currentDataset,
+            } for x in chunk]
+            containerReplicas += successItems
+            # Current algo will add files whole chunk, so total number of
+            # files in dataset is at most is max_file_per_datset+replicas_chunk_size.
+            #
+            # check the current number of files in the dataset
+            num = len(list(self.rucioClient.list_content(self.transfer.rucioScope, currentDataset)))
+            if num >= config.args.max_file_per_dataset:
+                # FIXME: close the last dataset when ALL Postjob has reach timeout.
+                self.rucioClient.close(self.transfer.rucioScope, currentDataset)
+                currentDataset = b.getOrCreateDataset(container)
+        return containerReplicas
+
+
+    def getSourcePFN(self, sourceLFN, sourceRSE, destinationRSE):
+        """
+        Get source PFN from `rucioClient.lfns2pfns()`.
+
+        :param sourceLFN: source LFN
+        :type sourceLFN: string
+        :param sourceRSE: source RSE where LFN is reside, but it must be normal
+            RSE name (e.g. `T2_CH_CERN` without suffix `_Temp`). Otherwise, it
+            will raise exception in `rucioClient.lfns2pfns()`.
+        :type sourceRSE: string
+        :param destinationRSE: need for select proper protocol for transfer
```

Comment:
I understand that if we want a source PFN, it will depend on destination for protocol choice. But this looks fragile and totally disgusting. Rucio should know how to move files. What will happen if one decides to move the rule while in progress to target a different destination ? 
Let's imagine that we have something similar to srm to webdav transition while this tool is in use ... will we survive ?
The very concept that we register a replica giving a PFN is somehow flawed. I can understand a path. But protocol ?
At the very least we need to make sure that there is a place where this thing is properly documented.
It looks the classic thing that "works now, but..." . I am starting to feel uncomfortable.
This may make better sense if we say: PFN for non-deterministic RSE's is always a WebDAV endpoint.
At least it makes it clear that if and when WebDav has to change, some serious thought is needed.
Anyhow this is not to criticize code here, if anything else, this code finally made the fragility apparent !

Reply:
after discussion we think that the bottom line is to acknowledge that
- the PFN stored in the Temp_RSE is not a general thing, but it is a PFN which can only be used for replicating this file to a specific destination.
Given that, current code is all in all clear enough, and forcing a 'davs' scheme would not make it any easier to understand.
All in all,, we expect replicas in Temp_RSE to only stay there for one month max.